test/tenant_token_test.dart

```diff
@@ -56,15 +56,16 @@ void main() {
         final admClient = MeiliSearchClient(testServer, admKey.key);
         final index = await createBooksIndex(uid: 'my_index');
         await index.updateFilterableAttributes(['tag', 'book_id']).waitFor(
-            client: client);
-
-        for (final Object rule in possibleRules) {
-          final token = admClient.generateTenantToken(admKey.uid!, rule);
-          final custom = MeiliSearchClient(testServer, token);
-
-          expect(() async => await custom.index('my_index').search(''),
-              returnsNormally);
-        }
+          client: client,
+        );
+
+        await Future.wait(
+          possibleRules.map((rule) {
+            final token = admClient.generateTenantToken(admKey.uid!, rule);
+            final custom = MeiliSearchClient(testServer, token);
+            return custom.index('my_index').search('');
```

Comment:
I read the explanation in the description, but don't we need the expect here? How will the `test` know there is a failure?

Reply:
the test will fail if there is an exception thrown anyway

Reply:
all `expect` is doing is throw an exception if the matcher isn't met